src/plugins/mdraid.c

```diff
@@ -1162,28 +1162,53 @@ gchar* bd_md_get_status (const gchar *raid_name, GError **error) {
 
 /**
  * bd_md_set_bitmap_location:
- * @raid_name: name of the RAID device to set the bitmap location
+ * @raid_name: name or path of the RAID device to set the bitmap location
  * @location: bitmap location (none, internal or path)
  * @error: (out): place to store error (if any)
  *
  * Returns: bitmap location for @raid_name RAID
  */
 gboolean bd_md_set_bitmap_location (const gchar *raid_name, const gchar *location, GError **error) {
+    const gchar *argv[] = {"mdadm", "--grow", NULL, "--bitmap", location, NULL};
+    gchar *md_path_str = NULL;
+    gchar *name_path_str = NULL;
     gboolean ret = FALSE;
-    gchar *md_path = g_strdup_printf ("/dev/md/%s", raid_name);
-    const gchar *argv[] = {"mdadm", "--grow", md_path, "--bitmap", location, NULL};
+
+    // XXX: This should be a separate function
+    if (g_str_has_prefix (raid_name, "/dev/"))
+        argv[2] = g_strdup (raid_name);
+    else
+      {
+        name_path_str = g_strdup_printf ("/dev/md/%s", raid_name);
+        md_path_str = g_strdup_printf ("/dev/%s", raid_name);
+        if (access (name_path_str, F_OK) == 0)
+            argv[2] = name_path_str;
+        else if (access (md_path_str, F_OK) == 0)
+            argv[2] = md_path_str;
+        else
+          {
```

Comment:
Please put the braces after ``else`` to the same line as the ``else`` itself. (you are not writing storaged code here ;-))